### ISSUE TITLE
kernel panic while accessing files from rclone mount

### ISSUE THREAD

**Issue by eXhumer** [osxfuse/osxfuse#982]
I had an encrypted remote mounted on my Macbook Air M1 with rclone, it was working fine until I tried accessing a file in my mount via SMB, which caused a kernal panic and rebooted the system. I'm on macOS Sonoma (14.0).

```
panic(cpu 0 caller 0xfffffe00232ab408): fuse: What kind of a flag is this (8100)?
Debugger message: panic
Memory ID: 0x6
OS release type: User
OS version: 23A344
Kernel version: Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:34 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T8103
Fileset Kernelcache UUID: 7CD3C0485714106B76F60CFF2DED0D9D
Kernel UUID: 9D9158F9-0E6D-30C0-BBA9-519175145599
Boot session UUID: 570940D5-E1E5-47F4-B658-45536E693CB4
iBoot version: iBoot-10151.1.1
secure boot?: YES
roots installed: 0
Paniclog version: 14
KernelCache slide: 0x000000001c378000
KernelCache base:  0xfffffe002337c000
Kernel slide:      0x000000001c380000
Kernel text base:  0xfffffe0023384000
Kernel text exec slide: 0x000000001d874000
Kernel text exec base:  0xfffffe0024878000
mach_absolute_time: 0x44ddca369
Epoch Time:        sec       usec
  Boot    : 0x65325df3 0x00002c45
  Sleep   : 0x00000000 0x00000000
  Wake    : 0x00000000 0x00000000
  Calendar: 0x653260ef 0x000e1937

Zone info:
  Zone map: 0xfffffe10003e0000 - 0xfffffe30003e0000
  . VM    : 0xfffffe10003e0000 - 0xfffffe14cd0ac000
  . RO    : 0xfffffe14cd0ac000 - 0xfffffe1666a44000
  . GEN0  : 0xfffffe1666a44000 - 0xfffffe1b33710000
  . GEN1  : 0xfffffe1b33710000 - 0xfffffe20003dc000
  . GEN2  : 0xfffffe20003dc000 - 0xfffffe24cd0a8000
  . GEN3  : 0xfffffe24cd0a8000 - 0xfffffe2999d74000
  . DATA  : 0xfffffe2999d74000 - 0xfffffe30003e0000
  Metadata: 0xfffffe3000bf0000 - 0xfffffe3008bf0000
  Bitmaps : 0xfffffe3008bf0000 - 0xfffffe300a1f0000
  Extra   : 0 - 0

CORE 0 recently retired instr at 0xfffffe0024a21840
CORE 1 recently retired instr at 0xfffffe0024a2303c
CORE 2 recently retired instr at 0xfffffe0024a2303c
CORE 3 recently retired instr at 0xfffffe0024a2303c
CORE 4 recently retired instr at 0xfffffe0024a2303c
CORE 5 recently retired instr at 0xfffffe0024a2303c
CORE 6 recently retired instr at 0xfffffe0024a2303c
CORE 7 recently retired instr at 0xfffffe0024a2303c
TPIDRx_ELy = {1: 0xfffffe1b35e62848  0: 0x0000000000000000  0ro: 0x000000016f5830e0 }
CORE 0 PVH locks held: None
CORE 1 PVH locks held: None
CORE 2 PVH locks held: None
CORE 3 PVH locks held: None
CORE 4 PVH locks held: None
CORE 5 PVH locks held: None
CORE 6 PVH locks held: None
CORE 7 PVH locks held: None
CORE 0 is the one that panicked. Check the full backtrace for details.
CORE 1: PC=0xfffffe0024a1df40, LR=0xfffffe0024a1df3c, FP=0xfffffe84fa75fe80
CORE 2: PC=0xfffffe0024a1df40, LR=0xfffffe0024a1df3c, FP=0xfffffe84fa537e80
CORE 3: PC=0xfffffe002490be14, LR=0xfffffe002490be14, FP=0xfffffe84fadebf20
CORE 4: PC=0xfffffe002490bc08, LR=0xfffffe002490bc08, FP=0xfffffe84f998bef0
CORE 5: PC=0xfffffe002490bc08, LR=0xfffffe002490bc08, FP=0xfffffe84fa243ef0
CORE 6: PC=0xfffffe002490bc08, LR=0xfffffe002490bc08, FP=0xfffffe84fabfbef0
CORE 7: PC=0xfffffe002490bc08, LR=0xfffffe002490bc08, FP=0xfffffe84f9c7fef0
Compressor Info: 5% of compressed pages limit (OK) and 3% of segments limit (OK) with 0 swapfiles and OK swap space
Panicked task 0xfffffe2997c7abe8: 951 pages, 3 threads: pid 1422: smbd
Panicked thread: 0xfffffe1b35e62848, backtrace: 0xfffffe84f9c1f2e0, tid: 19292
		  lr: 0xfffffe00248d1264  fp: 0xfffffe84f9c1f370
		  lr: 0xfffffe0024a29490  fp: 0xfffffe84f9c1f390
		  lr: 0xfffffe0024a1a4c8  fp: 0xfffffe84f9c1f400
		  lr: 0xfffffe0024a1893c  fp: 0xfffffe84f9c1f4f0
		  lr: 0xfffffe002487faa4  fp: 0xfffffe84f9c1f500
		  lr: 0xfffffe00248d0b44  fp: 0xfffffe84f9c1f8b0
		  lr: 0xfffffe00250e154c  fp: 0xfffffe84f9c1f8d0
		  lr: 0xfffffe00232ab408  fp: 0xfffffe84f9c1f9b0
		  lr: 0xfffffe00232a6ef4  fp: 0xfffffe84f9c1f9e0
		  lr: 0xfffffe0024ae9b98  fp: 0xfffffe84f9c1fad0
		  lr: 0xfffffe0024ad1118  fp: 0xfffffe84f9c1fd40
		  lr: 0xfffffe0024ad20a0  fp: 0xfffffe84f9c1fd90
		  lr: 0xfffffe0024ef6004  fp: 0xfffffe84f9c1fe20
		  lr: 0xfffffe0024a18aa4  fp: 0xfffffe84f9c1ff10
		  lr: 0xfffffe002487faa4  fp: 0xfffffe84f9c1ff20
      Kernel Extensions in backtrace:
         io.macfuse.filesystems.macfuse(2128.20)[BBB320C0-222A-3582-85BE-3C47BA8FAD36]@0xfffffe00232a0000->0xfffffe00232b2b73

last started kext at 1975446919: io.macfuse.filesystems.macfuse	2128.20 (addr 0xfffffe0023298000, size 6320)
loaded kexts:
io.macfuse.filesystems.macfuse	2128.20
com.apple.filesystems.autofs	3.0
com.apple.driver.AppleTopCaseHIDEventDriver	7400.26
com.apple.driver.CoreKDL	1
com.apple.driver.AppleBiometricServices	1
com.apple.driver.AppleALSColorSensor	1.0.0d1
com.apple.driver.AppleAOPVoiceTrigger	300.7
com.apple.driver.BCMWLANFirmware4378.Hashstore	1
com.apple.driver.DiskImages.ReadWriteDiskImage	493.0.0
com.apple.driver.DiskImages.UDIFDiskImage	493.0.0
com.apple.driver.DiskImages.RAMBackingStore	493.0.0
com.apple.driver.DiskImages.FileBackingStore	493.0.0
com.apple.driver.SEPHibernation	1
com.apple.driver.AppleSmartBatteryManager	161.0.0
com.apple.driver.AppleUSBDeviceNCM	5.0.0
com.apple.driver.AppleThunderboltIP	4.0.3
com.apple.driver.AppleFileSystemDriver	3.0.1
com.apple.nke.l2tp	1.9
com.apple.filesystems.tmpfs	1
com.apple.driver.AppleSmartIO2	1
com.apple.AppleEmbeddedSimpleSPINORFlasher	1
com.apple.driver.ApplePMP	1
com.apple.filesystems.nfs	1
com.apple.filesystems.lifs	1
com.apple.filesystems.apfs	2235.0.13
com.apple.IOTextEncryptionFamily	1.0.0
com.apple.filesystems.hfs.kext	650.0.2
com.apple.security.BootPolicy	1
com.apple.BootCache	40
com.apple.AppleFSCompression.AppleFSCompressionTypeZlib	1.0.0
com.apple.AppleFSCompression.AppleFSCompressionTypeDataless	1.0.0d1
com.apple.driver.ApplePMPFirmware	1
com.apple.driver.AppleSerialShim	1
com.apple.driver.AudioDMAController-T8103	300.15
com.apple.driver.AppleEventLogHandler	1
com.apple.driver.AppleTAS5770LAmp	700.46
com.apple.driver.AppleDPDisplayTCON	1
com.apple.driver.AppleJPEGDriver	6.1.1
com.apple.driver.AppleCS42L83Audio	700.46
com.apple.driver.AppleAVE2	703.6.9
com.apple.driver.AppleAVD	732
com.apple.driver.AppleT8020SOCTuner	1
com.apple.driver.AppleT8103CLPCv3	1
com.apple.driver.AppleT8103PMGR	1
com.apple.driver.AppleS5L8960XNCO	1
com.apple.driver.AppleMobileDispH13G-DCP	140.0
com.apple.AGXG13G	275.6
com.apple.driver.usb.AppleSynopsysUSB40XHCI	1
com.apple.driver.AppleInterruptController	1.0.0d1
com.apple.driver.AppleS8000DWI	1.0.0d1
com.apple.driver.AppleS8000AES	1
com.apple.driver.AppleT8020DART	1
com.apple.driver.AppleBluetoothModule	1
com.apple.driver.AppleSamsungSerial	1.0.0d1
com.apple.driver.AppleBCMWLANBusInterfacePCIe	1
com.apple.driver.AppleS5L8920XPWM	1.0.0d1
com.apple.driver.AppleS5L8940XI2C	1.0.0d2
com.apple.driver.AppleSPIMC	1
com.apple.driver.AppleT8101	1
com.apple.driver.AppleM68Buttons	1.0.0d1
com.apple.iokit.IOUserEthernet	1.0.1
com.apple.driver.usb.AppleUSBUserHCI	1
com.apple.iokit.IOKitRegistryCompatibility	1
com.apple.iokit.EndpointSecurity	1
com.apple.driver.AppleDiskImages2	272.0.0
com.apple.AppleSystemPolicy	2.0.0
com.apple.nke.applicationfirewall	404
com.apple.kec.InvalidateHmac	1
com.apple.kec.AppleEncryptedArchive	1
com.apple.driver.driverkit.serial	6.0.0
com.apple.iokit.IOAVBFamily	1200.18
com.apple.driver.AppleHSBluetoothDriver	7400.26
com.apple.driver.IOBluetoothHIDDriver	9.0.0
com.apple.driver.AppleHIDKeyboard	7400.2
com.apple.driver.AppleActuatorDriver	7400.42
com.apple.driver.AppleMultitouchDriver	7400.42
com.apple.driver.usb.IOUSBHostHIDDevice	1.2
com.apple.driver.usb.cdc	5.0.0
com.apple.driver.AppleUSBAudio	600.40
com.apple.iokit.IOAudioFamily	500.4
com.apple.vecLib.kext	1.2.0
com.apple.driver.AppleMesaSEPDriver	100.99
com.apple.iokit.IOBiometricFamily	1
com.apple.driver.AppleAOPAudio	300.14
com.apple.driver.DiskImages.KernelBacked	493.0.0
com.apple.driver.AppleSEPHDCPManager	1.0.1
com.apple.driver.AppleTrustedAccessory	1
com.apple.iokit.AppleSEPGenericTransfer	1
com.apple.driver.AppleConvergedIPCOLYBTControl	1
com.apple.driver.AppleConvergedPCI	1
com.apple.driver.AppleBluetoothDebug	1
com.apple.driver.AppleBTM	1.0.1
com.apple.driver.AppleXsanScheme	3
com.apple.driver.usb.networking	5.0.0
com.apple.driver.AppleThunderboltPCIDownAdapter	4.1.1
com.apple.driver.AppleThunderboltDPInAdapter	8.5.1
com.apple.driver.AppleThunderboltDPAdapterFamily	8.5.1
com.apple.driver.AppleThunderboltUSBDownAdapter	1.0.4
com.apple.nke.ppp	1.9
com.apple.driver.AppleHIDTransportSPI	7400.34
com.apple.driver.AppleHIDTransport	7400.34
com.apple.driver.AppleInputDeviceSupport	7400.34
com.apple.driver.AppleH13CameraInterface	8.24.1
com.apple.driver.AppleH11ANEInterface	7.23.0
com.apple.driver.AppleDiagnosticDataAccessReadOnly	1.0.0
com.apple.driver.AppleStockholmControl	1.0.0
com.apple.driver.AppleDCPDPTXProxy	1.0.0
com.apple.driver.DCPDPFamilyProxy	1
com.apple.driver.AppleBSDKextStarter	3
com.apple.kext.triggers	1.0
com.apple.driver.IOHIDPowerSource	1
com.apple.driver.AppleCallbackPowerSource	1
com.apple.filesystems.hfs.encodings.kext	1
com.apple.driver.AppleSyntheticGameController	11.0.24
com.apple.driver.AppleHPM	3.4.4
com.apple.driver.AppleT8103TypeCPhy	1
com.apple.driver.AppleSPU	1
com.apple.driver.AppleSART	1
com.apple.iokit.IONVMeFamily	2.1.0
com.apple.driver.AppleNANDConfigAccess	1.0.0
com.apple.AGXFirmwareKextG13GRTBuddy	275.6
com.apple.AGXFirmwareKextRTBuddy64	275.6
com.apple.driver.AppleSPMIPMU	1.0.1
com.apple.driver.AppleDialogPMU	1.0.1
com.apple.driver.AppleCSEmbeddedAudio	700.46
com.apple.driver.AppleEmbeddedAudio	700.46
com.apple.iokit.AppleARMIISAudio	300.11
com.apple.plugin.IOgPTPPlugin	1200.91
com.apple.driver.ApplePassthroughPPM	3.0
com.apple.driver.ApplePMGR	1
com.apple.iokit.IOMobileGraphicsFamily-DCP	343.0.0
com.apple.iokit.IOMobileGraphicsFamily	343.0.0
com.apple.driver.AppleDCP	1
com.apple.driver.DCPAVFamilyProxy	1
com.apple.driver.AppleFirmwareKit	1
com.apple.driver.AppleM2ScalerCSCDriver	265.0.0
com.apple.iokit.IOGPUFamily	93
com.apple.driver.AppleUSBXDCIARM	1.0
com.apple.driver.AppleUSBXDCI	1.0
com.apple.iokit.IOUSBDeviceFamily	2.0.0
com.apple.driver.usb.AppleSynopsysUSBXHCI	1
com.apple.driver.usb.AppleUSBXHCI	1.2
com.apple.driver.AppleEmbeddedUSBHost	1
com.apple.driver.usb.AppleUSBHub	1.2
com.apple.driver.usb.AppleUSBHostCompositeDevice	1.2
com.apple.driver.AppleSPMI	1.0.1
com.apple.driver.AppleARMWatchdogTimer	1
com.apple.driver.AppleDisplayCrossbar	1.0.0
com.apple.iokit.IODisplayPortFamily	1.0.0
com.apple.driver.AppleTypeCPhy	1
com.apple.driver.AppleThunderboltNHI	7.2.81
com.apple.driver.AppleT8103PCIeC	1
com.apple.iokit.IOThunderboltFamily	9.3.3
com.apple.iokit.IOPortFamily	1.0
com.apple.driver.ApplePIODMA	1
com.apple.driver.AppleT8103PCIe	1
com.apple.driver.AppleMultiFunctionManager	1
com.apple.driver.AppleBluetoothDebugService	1
com.apple.driver.AppleBCMWLANCore	1.0.0
com.apple.iokit.IO80211Family	1200.13.0
com.apple.driver.IOImageLoader	1.0.0
com.apple.driver.AppleOLYHAL	1
com.apple.driver.corecapture	1.0.4
com.apple.driver.AppleEmbeddedPCIE	1
com.apple.driver.AppleMCA2-T8103	800.11
com.apple.driver.AppleEmbeddedAudioLibs	300.1
com.apple.driver.AppleFirmwareUpdateKext	1
com.apple.driver.AppleGPIOICController	1.0.2
com.apple.driver.AppleFireStormErrorHandler	1
com.apple.driver.AppleMobileApNonce	1
com.apple.driver.usb.AppleUSBHostPacketFilter	1.0
com.apple.iokit.IOTimeSyncFamily	1200.91
com.apple.driver.DiskImages	493.0.0
com.apple.iokit.IOGraphicsFamily	597
com.apple.iokit.IOBluetoothFamily	9.0.0
com.apple.driver.AppleSSE	1.0
com.apple.driver.AppleSEPKeyStore	2
com.apple.driver.AppleUSBTDM	556
com.apple.iokit.IOUSBMassStorageDriver	240
com.apple.iokit.IOPCIFamily	2.9
com.apple.iokit.IOSCSIBlockCommandsDevice	492
com.apple.iokit.IOSCSIArchitectureModelFamily	492
com.apple.driver.AppleRSMChannel	1
com.apple.iokit.IORSMFamily	1
com.apple.driver.AppleLockdownMode	1
com.apple.driver.AppleIPAppender	1.0
com.apple.driver.AppleFDEKeyStore	28.30
com.apple.driver.AppleEffaceableStorage	1.0
com.apple.driver.AppleCredentialManager	1.0
com.apple.driver.KernelRelayHost	1
com.apple.iokit.IOUSBHostFamily	1.2
com.apple.driver.AppleUSBHostMergeProperties	1.2
com.apple.driver.usb.AppleUSBCommon	1.0
com.apple.driver.AppleSMC	3.1.9
com.apple.driver.RTBuddy	1.0.0
com.apple.driver.AppleEmbeddedTempSensor	1.0.0
com.apple.driver.AppleARMPMU	1.0
com.apple.iokit.IOAccessoryManager	1.0.0
com.apple.driver.AppleUVDM	1.0.0
com.apple.driver.AppleOnboardSerial	1.0
com.apple.iokit.IOSkywalkFamily	1.0
com.apple.driver.mDNSOffloadUserClient	1.0.1b8
com.apple.iokit.IONetworkingFamily	3.4
com.apple.iokit.IOSerialFamily	11
com.apple.driver.AppleSEPManager	1.0.1
com.apple.driver.IODARTFamily	1
com.apple.driver.AppleA7IOP	1.0.2
com.apple.driver.IOSlaveProcessor	1
com.apple.driver.AppleBiometricSensor	2
com.apple.iokit.IOHIDFamily	2.0.0
com.apple.AUC	1.0
com.apple.iokit.IOSurface	352.0.3
com.apple.iokit.IOAVFamily	1.0.0
com.apple.iokit.IOHDCPFamily	1.0.0
com.apple.iokit.IOCECFamily	1
com.apple.iokit.IOAudio2Family	1.0
com.apple.driver.AppleIISController	300.1
com.apple.driver.AppleAudioClockLibs	300.1
com.apple.driver.FairPlayIOKit	71.3.0
com.apple.driver.AppleARMPlatform	1.0.2
com.apple.iokit.IOSlowAdaptiveClockingFamily	1.0.0
com.apple.iokit.IOReportFamily	47
com.apple.security.quarantine	4
com.apple.security.sandbox	300.0
com.apple.iokit.IOStorageFamily	2.1
com.apple.kext.AppleMatch	1.0.0d1
com.apple.driver.AppleMobileFileIntegrity	1.0.5
com.apple.iokit.CoreAnalyticsFamily	1
com.apple.security.AppleImage4	5.0.0
com.apple.kext.CoreTrust	1
com.apple.iokit.IOCryptoAcceleratorFamily	1.0.1
com.apple.kec.pthread	1
com.apple.kec.Libm	1
com.apple.kec.Compression	1.0
com.apple.kec.corecrypto	14.0



** Stackshot Succeeded ** Bytes Traced 427243 (Uncompressed 1081952) **
```

**Comment by eXhumer** [comment]
There is no kernal panic if I access any file on the remote locally on the Macbook, but trying to access the same file over SMB is causing kernal panics everytime.

**Comment by bfleischer** [comment]
SMB seems to be passing an undocumented flag to macFUSE that is not handled correctly.

**Comment by bfleischer** [comment]
The issue is fixed in the upcoming 4.6.0 release.
